include/bluetooth/conn.h

```diff
@@ -932,6 +932,17 @@ struct bt_conn_cb {
  */
 void bt_conn_cb_register(struct bt_conn_cb *cb);
 
+/** @def BT_CONN_CB_DEFINE
+ *
+ *  @brief Register a callback structure for connection events.
+ *
+ *  @param _name Name of callback structure.
+ */
+#define BT_CONN_CB_DEFINE(_name)					\
+	static const Z_STRUCT_SECTION_ITERABLE(bt_conn_cb,		\
+						_CONCAT(bt_conn_cb,	\
+							_name))
```

Comment:
Is this really the common or preferred style for this kind of macros, where you leave it up to the user to provide the struct definition after the macro?

I'd have thought that it's more common to provide the given struct members as parameters to the macro. Also, I think providing a variable name as input is irrelevant since it's not needed anywhere in the code. Instead you can generate unique names using e.g. `__COUNTER__`. So the usage of the macro would then look simply like:
```
BT_CONN_CB_DEFINE(connected, disconnected);
```

Reply:
That said, I do see the advantage of leaving the definition details to the user, since it does produce code that's more readable, and also covers the case you pointed out earlier where you have the need to have some members build-time conditional.

Reply:
Relevant conversation: https://github.com/zephyrproject-rtos/zephyr/pull/28054#issuecomment-715168117